br/tests/br_charset_gbk/run.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+#
+# Copyright 2022 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+check_cluster_version 5 4 0 'new collation' || { echo 'TiDB does not support new collation! skipping test'; exit 0; }
+
+set -eu
+
+cur=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+. $cur/../_utils/run_services
+
+# restart cluster with new collation enabled
+start_services --tidb-cfg $cur/tidb-new-collation.toml
+
+DB="$TEST_NAME"
+PROGRESS_FILE="$TEST_DIR/progress_unit_file"
+rm -rf $PROGRESS_FILE
+
+run_sql "CREATE DATABASE $DB;"
+
+run_sql "CREATE TABLE $DB.测试 ( \
+    a char(20) DEFAULT NULL, \
+    b tinyblob, \
+    c binary(100) DEFAULT NULL, \
+    d json DEFAULT NULL, \
+    e timestamp NULL DEFAULT NULL, \
+    f set('a一','b二','c三','d四') DEFAULT NULL, \
+    g text, \
+    h enum('a一','b二','c三','d四') DEFAULT 'c三' \
+) ENGINE=InnoDB DEFAULT CHARSET=gbk COLLATE=gbk_chinese_ci;"
+
+run_sql "INSERT INTO $DB.测试 VALUES ('你好', '你好', '你好', '{\"测试\": \"你好\"}', '2018-10-13', 1, '你好', 'a一');"
+run_sql "INSERT INTO $DB.测试 VALUES ('你好123', '你好', '你好', '{\"测试\": \"你好\"}', '2018-10-13', 1, '你好', 'a一');"
+
+run_sql "CREATE TABLE $DB.t ( \
+  YCSB_KEY varchar(64) NOT NULL, \
+  FIELD0 varchar(1) DEFAULT NULL, \
+  PRIMARY KEY (YCSB_KEY) \
+) ENGINE=InnoDB DEFAULT CHARSET=gbk;"
+
+run_sql "INSERT INTO $DB.t VALUES (\"测试\", \"你\");"
+run_sql "SET NAMES gbk; INSERT INTO $DB.t VALUES (\"测试\", \"a\"); SET NAMES default;"
+
+# backup db
+echo "backup start..."
+export GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/task/progress-call-back=return(\"$PROGRESS_FILE\")"
```

Comment:
we can remove `PROGRESS_FILE` failpoint in this test

Reply:
Done. PTAL @3pointer